index.js

```diff
@@ -1,67 +1,53 @@
 'use strict';
 var condense = require('selective-whitespace');
-var eachAsync = require('each-async');
 var netstats = require('netstats');
 var platform = process.platform;
 require('native-promise-only');
 
-var pids = {
-	all: [],
-	tcp: [],
-	udp: []
-};
-
 module.exports = function (port, opts) {
 	if (typeof port !== 'number') {
 		throw new TypeError('Expected a port number');
 	}
 
 	opts = opts || {};
 
-	return new Promise(function (resolve) {
-		netstats(port).then(function (stats) {
-			process(stats).then(function (ps) {
-				resolve(ps);
-			});
-		}).catch(function () {
-			resolve(pids);
-		});
-	});
+	return netstats(port).then(processNetStats);
 };
 
-function process(arr) {
+function processNetStats(arr) {
 	var pidindex = 1;
 	var items = arr.slice(1);
 
 	if (platform === 'win32') {
 		items = arr;
 	}
 
-	return new Promise(function (resolve, reject) {
-		eachAsync(items, function (item, index, done) {
-			var values = condense(item).split(' ');
-			var pid = parseInt(values[pidindex], 10);
+	var pids = {
+		all: [],
+		tcp: [],
+		udp: []
+	};
 
-			if (platform === 'win32') {
-				pid = values.pop();
-			}
+	return Promise.all(items.map(function (item) {
```

Comment:
I don't see any promises being returned in this `Promise.all()`. Where are the promises?